sql/hive/src/main/scala/org/apache/spark/sql/hive/security/HiveDelegationTokenProvider.scala

```diff
@@ -99,7 +99,18 @@ private[spark] class HiveDelegationTokenProvider
         s"$principal at $metastoreUri")
 
       doAsRealUser {
-        val hive = Hive.get(conf, classOf[HiveConf])
+        val hive = conf match {
+          case hiveConf: HiveConf =>
+            try {
```

Comment:
can we reduce the code duplication by moving `HiveClientImpl.getHive` to its companion object and then call the method here?

Reply:
done